test/conftest.py

```diff
@@ -104,9 +111,10 @@ def pytest_sessionstart(session: pytest.Session) -> None:
 
     # Run stuff just once for the pytest session even running under xdist.
     if "xdist" not in sys.modules or not sys.modules["xdist"].is_xdist_worker(request_or_session=session):
-        temp_dir = os.path.join(session.config.rootpath, "..", session.config.getoption("--tmpdir"))
+        global tp_server, finalize
+        temp_dir = Path(session.config.getoption("--tmpdir")).absolute()
         prepare_dirs(tempdir_base=temp_dir, modes=session.config.getoption("--mode") or get_configured_modes())
-        start_s3_mock_services(minio_tempdir_base=temp_dir)
+        start_3rd_party_services(tempdir_base=temp_dir, byte_limit=session.config.getoption('byte_limit'))
```

Comment:
I've been saying this again and again, but I'll say it yet again (but as usual, won't insist):

Different tests need different selections of "3rd party services". Some will need s3, some need ldap, some (?) need both and some neither. When we had this big do-everything-in-one-place test.py, it was fine that it started all services. But now, it's possible to run with pytest a single test/alternator (for example) test function. Why should we run s3 or ldap in this case???

I will not insist because you do not run all this stuff if the "--pytest-init" option isn't used so I can still run things without it, so I (and the way I recommend to others to quickly run tests) will not get hurts. But I can still point out that in the long run, this is a design mistake.